src/unix/udp.c

```diff
@@ -645,7 +645,7 @@ int uv__udp_connect(uv_udp_t* handle,
     err = connect(handle->io_watcher.fd, addr, addrlen);
   } while (err == -1 && errno == EINTR);
 
-  if (err)
+  if (err && errno != EINPROGRESS)
     return UV__ERR(errno);
```

Comment:
I think this would fix the issue in most of the cases but I think we can do better. The man connect page states:

> After select(2) indicates writability, use getsockopt(2) to read the SO_ERROR option at level SOL_SOCKET to determine whether connect() completed successfully (SO_ERROR is zero) or unsuccessfully (SO_ERROR is one of the usual error codes listed here, explaining the reason for the failure).

So, what I'd suggest is setting a specific flag (for example `UV_HANDLE_UDP_CONNECT_IN_PROGRESS`) here and in `uv__udp_sendmsg()`, if it's set, check `SO_ERROR` to see if it's already connected (remove this flag and set `UV_HANDLE_UDP_CONNECTED`) and we can continue. If it still returns `EINPROGRESS`, we should call `uv__io_start()` with `POLLOUT` and when the socket is notified for writability we'll know whether the socket is connected or not.

Reply:
You are right, but  check for `UV_HANDLE_UDP_CONNECT_IN_PROGRESS` have to be in `uv__udp_check_before_send`

Reply:
Done, please review PR